lang/java/avro/src/main/java/org/apache/avro/Conversions.java

```diff
@@ -44,6 +44,17 @@
 public class Conversions {
 
   public static class UUIDConversion extends Conversion<UUID> {
+
+    private final boolean isBigEndian;
+
+    public UUIDConversion() {
+      this(ByteOrder.nativeOrder() == ByteOrder.BIG_ENDIAN);
```

Comment:
I'm not sure about this. I would say that we always write BigEndian, similar to Iceberg: https://github.com/apache/iceberg/blob/main/api/src/main/java/org/apache/iceberg/util/UUIDUtil.java#L76-L79

Reply:
Given that UUIDs are 128 bit labels (source with extra references: https://en.wikipedia.org/wiki/Universally_unique_identifier#Binary_wire_format), I think this is the best choice.

Reply:
Having done a bit oof research, network byte order is the way to go. For reference: [RFC4122, section 4.12.2](https://www.ietf.org/rfc/rfc4122.html#section-4.1.2).